src/heap/CustomAllocator.cpp

```diff
@@ -197,6 +198,25 @@ int getValidValueInWeakRefObject(void* ptr, GC_mark_custom_result* arr)
     arr[2].to = (GC_word*)current->m_values.data();
     return 0;
 }
+
+int getValidValueInFinalizationRegistryObjectItem(void* ptr, GC_mark_custom_result* arr)
+{
+    FinalizationRegistryObject::FinalizationRegistryObjectItem* current = (FinalizationRegistryObject::FinalizationRegistryObjectItem*)ptr;
+    arr[0].from = (GC_word*)&current->heldValue;
```

Comment:
`getValidValueInFinalizationRegistryObjectItem`  marks only `heldValue` and `unregisterToken` members only.
It seems that `source` member needs to be marked too.

Reply:
Oops! you are right